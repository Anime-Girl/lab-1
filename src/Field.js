import React from 'react'

class Field extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            id: props.id,
            value: props.value,
            actSubmit: props.actSubmit,
            //actChange: props.actChange,
        }
        this.onClick = this.onClick.bind(this)
        this.onChange = this.onChange.bind(this)

    }
    onChange(val) {
        this.setState({ value: val });
    }

    onClick = () => {
        this.state.actSubmit(this.state.value);
    }

    render() {
        return (
            <div class="button" className='Field' >
                Color:
                <input type="text" id={this.state.id} value={this.state.value} onChange={e => this.onChange(e.target.value)} />
                <button onClick={this.onClick} type="submit" value="Submit"> </button>
            </div>
        )
    }
}
export default Field;