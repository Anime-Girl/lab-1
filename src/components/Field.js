import React from 'react'

class Field extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            id: props.id,
            value: props.value,
            actSubmit: props.actSubmit,
        }
        this.onClick = this.onClick.bind(this)
        this.onChange = this.onChange.bind(this)
    }
    onChange(val) {
        this.setState({ value: val });
    }
    onClick = () => {
        this.state.actSubmit(this.state.value);
    }
    render() {
        return (
            <div class="button" className='Field' >
                Color:
                <input id={this.state.id} value={this.state.value} onChange={e => this.onChange(e.target.value)} />
                <button id="ourbutton" onClick={this.onClick} type="submit" value="Submit">content</button>
            </div >
        )
    }
}
export default Field;