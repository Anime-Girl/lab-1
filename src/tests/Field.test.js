import React, { createElement } from 'react';
import { create } from 'react-test-renderer';
import Field from '../components/Field';

// Still makes sense to see if this Component actually renderable
describe('[RENDER] Field â€“ Snapshot test', () => {
  test('Matches the snapshot', () => {
    const field = create(<Field />);
    expect(field.toJSON()).toMatchSnapshot();
  });
});
