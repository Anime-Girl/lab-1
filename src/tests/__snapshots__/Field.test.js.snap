// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Field â€“ Snapshot test Matches the snapshot 1`] = `
<div
  class="button"
  className="Field"
>
  Color:
  <input
    onChange={[Function]}
  />
  <button
    id="ourbutton"
    onClick={[Function]}
    type="submit"
    value="Submit"
  >
    content
  </button>
</div>
`;
